### DISCUSSION TITLE
Why is __NEXT_INIT_URL not exposed?

### DISCUSSION THREAD

**Discussion by Haroenv** [vercel/next.js#44527] [Ideas]
If in some cases on the browser you use `window.location.href` in a link which gets dynamically created (maybe with URL constructor, which only accepts full URLs), on the backend you can pass `url` through your app in a provider so you can use the same exact URL everywhere.

It seems like this is the base of parsedURL, but that URL is relative, and thus not the same as `location.href` for example: 

https://github.com/vercel/next.js/blob/2c9c8f76a36f919bae8e10d7965463080f10e277/packages/next/server/base-server.ts#L1150

Currently I use weird workarounds like:

```js
  async function getServerSideProps({ req }) {
    const protocol = req.headers.referer?.split('://')[0] || 'http';
    const url = `${protocol}://${req.headers.host}${req.url}`;
  }
```

Would it make sense to expose `__NEXT_INIT_URL` instead?

**Comment by tomascasas** [comment]
Specially since that example is still missing the `port` which is part of the `origin`.
